Combination-sum-leetcode-problem.py

```diff
@@ -0,0 +1,23 @@
+class Solution:
+    def combinationSum(self, candidates: List[int], target: int) -> List[List[int]]:
+        self.ans = []
+        self.target = target
+        self.backTracking(candidates, [], 0, 0)
+        
+        return self.ans
+    
+    #backtracking
+    def backTracking(self, candidates, path, idx, add):
```

Comment:
The recursive backtracking function is implemented correctly and it has simple and understandable base case. This is good practice